pkgs/development/libraries/livesplit-core/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+, dynamicLib ? true
+, generateBindings ? true
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "livesplit-core";
+  version = "unstable-2021-08-23";
+
+  src = fetchFromGitHub {
+    owner = "livesplit";
+    repo = pname;
+    rev = "6fb02347f0ec4db8fb4bd8e284f8a8bc71e15686";
+    sha256 = "0279hmlcdpbd5g6x02vp8gq8zj8j37kzw6d5k892rp5fr4z5xkga";
+  };
+
+  cargoPatches = [ ./Cargo.lock ];
+  cargoSha256 = "1g3x9aaq108lnk1yvyhqkwyzwfxias12z2bkbi8s8qyk3z3vgg22";
```

Comment:
> Can you add a comment explaining why we need to do this patching so we can know if we're able to remove it in the future?

I think it should be fairly self explanatory, as the filename pretty much sums it up. But more documentation can never hurt, I'll add the comment :)

<br />

> Is there a better way to handle a non-rust application that depends on a rust library?

If you're refering to redtimer [symlinking some bindings script and an shared object](https://github.com/NixOS/nixpkgs/pull/135479/files#diff-476b92c05b7d91ff7f0aa5b177386f644384ff6f89ab61ecb2f9c91107c1ec44R31) to the projects source, that's just because redtimer's way of loading stuff isn't the best for portability. 

This is what you're meant to do, according to their [build instructions](https://github.com/cout/redtimer#build-instructions).  Also notice [how you have to be in the projects root directory](https://github.com/NixOS/nixpkgs/pull/135479/files#diff-476b92c05b7d91ff7f0aa5b177386f644384ff6f89ab61ecb2f9c91107c1ec44R44) in order for it to find it's other modules, including the symlinked files from `livesplit-core`. 

Packages focused more on portability could load these in a much more elegant way.

Reply:
What you've done here seems reasonable, given the complexity of packaging something that needs both ruby and rust locks.

Reply:
Thank you for taking a look :)